From 6e9d919c070f46935203fccc1fa114a0f1fb8592 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 12 Apr 2022 14:25:10 +0200
Subject: [PATCH 0151/1695] clk: stm32mp13: add adc bus clocks

Bus clock is mandatory as also needed for register interface.
'adc' clock is optional: it's the one used for analog when specified
in the device tree.

Current clock driver for MP13 exposes only ADC kernel clock.
There's an issue then for clock gating when only the BUS clock
is used for both registers access and analog.
Then we have to manage a counter at the gate level.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9453f4b0181576c8d03ec0edf3297fb0c687aceb clk: stm32mp13: add adc bus clocks]
Change-Id: I70556a98c3de24a7a509912b1dae0600291493fb
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/drivers/clk/clk-stm32mp13.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/drivers/clk/clk-stm32mp13.c b/core/drivers/clk/clk-stm32mp13.c
index 308256ac7..b4c37cc58 100644
--- a/core/drivers/clk/clk-stm32mp13.c
+++ b/core/drivers/clk/clk-stm32mp13.c
@@ -2169,6 +2169,8 @@ static  STM32_GATE(ck_tim17_k, &ck_timg3, 0, GATE_TIM17);
 static  STM32_GATE(ck_ltdc_px, &ck_pll4q, 0, GATE_LTDC);
 static  STM32_GATE(ck_dma1, &ck_mlahb, 0, GATE_DMA1);
 static  STM32_GATE(ck_dma2, &ck_mlahb, 0, GATE_DMA2);
+static  STM32_GATE(ck_adc1, &ck_mlahb, 0, GATE_ADC1);
+static  STM32_GATE(ck_adc2, &ck_mlahb, 0, GATE_ADC2);
 static  STM32_GATE(ck_mdma, &ck_axi, 0, GATE_MDMA);
 static  STM32_GATE(ck_eth1mac, &ck_axi, 0, GATE_ETH1MAC);
 static  STM32_GATE(ck_usbh, &ck_axi, 0, GATE_USBH);
@@ -2483,6 +2485,8 @@ static struct clk *stm32mp13_clk_provided[STM32MP13_ALL_CLK_NB] = {
 	[LTDC_PX]	= &ck_ltdc_px,
 	[DMA1]		= &ck_dma1,
 	[DMA2]		= &ck_dma2,
+	[ADC1]		= &ck_adc1,
+	[ADC2]		= &ck_adc2,
 	[MDMA]		= &ck_mdma,
 	[ETH1MAC]	= &ck_eth1mac,
 	[USBH]		= &ck_usbh,
-- 
2.47.2

